MergerLogic/ImageProcessing/TileScaler.cs

```diff
@@ -9,25 +9,52 @@ public class TileScaler : ITileScaler
         private const int TILE_WIDTH = 256;
         private const int TILE_HEIGHT = 256;
 
-        public void Upscale(MagickImage baseImage, Tile baseTile, Coord targetCoords)
+        public MagickImage Upscale(MagickImage baseImage, Tile baseTile, Coord targetCoords)
         {
             int zoomLevelDiff = targetCoords.Z - baseTile.Z;
             int scale = 1 << zoomLevelDiff;
-            double scaleAsDouble = (double)scale;
 
-            double tilePartX = (targetCoords.X % scale) / scaleAsDouble;
-            double tilePartY = (targetCoords.Y % scale) / scaleAsDouble;
-            double tileSize = TILE_HEIGHT / scaleAsDouble;
+            //find source pixels range
+            double tilePartX = targetCoords.X % scale;
+            double tilePartY = targetCoords.Y % scale;
+            double tileSize = TILE_HEIGHT / (double)scale;
 
-            int pixleX = (int)(tilePartX * TILE_WIDTH);
-            int pixleY = (int)(tilePartY * TILE_HEIGHT);
-            int imageWidth = Math.Max((int)tileSize, 1);
-            int imageHeight = Math.Max((int)tileSize, 1);
+            int pixelX = (int)(tilePartX * tileSize);
```

Comment:
This is incorrect because of my last comment, it will not give you a value in the correct bounds.

Reply:
it is not because of my answer to your last comment